source/extensions/clusters/redis/redis_cluster.cc

```diff
@@ -225,22 +227,14 @@ RedisCluster::RedisDiscoverySession::RedisDiscoverySession(
 // Convert the cluster slot IP/Port response to and address, return null if the response
 // does not match the expected type.
 Network::Address::InstanceConstSharedPtr
-RedisCluster::RedisDiscoverySession::RedisDiscoverySession::ProcessCluster(
+RedisCluster::RedisDiscoverySession::RedisDiscoverySession::processClusterByIP(
     const NetworkFilters::Common::Redis::RespValue& value) {
-  if (value.type() != NetworkFilters::Common::Redis::RespType::Array) {
-    return nullptr;
-  }
-  auto& array = value.asArray();
-
-  if (array.size() < 2 || array[0].type() != NetworkFilters::Common::Redis::RespType::BulkString ||
-      array[1].type() != NetworkFilters::Common::Redis::RespType::Integer) {
-    return nullptr;
-  }
 
+  const auto& array = value.asArray();
   try {
     return Network::Utility::parseInternetAddress(array[0].asString(), array[1].asInteger(), false);
```

Comment:
Maybe use `parseInternetAddressNoThrow` and handle null? seems a bit clearer perhaps than catching the exception.